metals/src/main/scala/scala/meta/internal/metals/ScalaVersions.scala

```diff
@@ -84,7 +84,7 @@ object ScalaVersions {
     if (isScala3Milestone(scalaVersion))
       scalaVersion
     else
-      scalaVersion.split('.').take(2).mkString(".")
+      scalaVersion.split('.').take(2).takeWhile(_ != "0").mkString(".")
```

Comment:
3.0.0 needs to translate to 3

Reply:
this will break with 3.1.0. I realize it's far away, but maybe we can be a bit more explicit here:

```scala
if (isScala3Milestone(scalaVersion))
  scalaVersion
else if (scalaVersion.startsWith("3"))
  "3"
else
  scalaVersion.split(".").take(2).mkString(".")
```

?

Reply:
Why would it break for `3.1.0` ? It should return `3.1`, which I think would be expected.

Reply:
what's this version used for? If 3.0.0 returns 3, I think 3.1.0 should return 3 as well

Reply:
That's the binary version, so I think we will break binary compatibility for 3.1, or will it be always "3" ?

Reply:
my understanding is that it will always be "3" due to the new binary compatibility guarantees in Scala 3. Minor versions of the language won't break binary compatibility (concretely we will be able to use libraries compiled for 3.0 from 3.1)

Reply:
You are right, that's pretty neat! Will change it then.

Reply:
Fixed!